flux-sched: set the version if the ver file is missing

trws opened the pull request:
problem: flux-sched needs a version, it normally gets this from a release tarball or from git tags, but if using a source archive or a git clone without tags the version is missing

solution: set the version through cmake based on the version spack sees when the version file is missing

trws committed:
flux-sched: set the version if the ver file is missing

problem: flux-sched needs a version, it normally gets this from a
release tarball or from git tags, but if using a source archive or a git
clone without tags the version is missing

solution: set the version through cmake based on the version spack sees
when the version file is missing

grondo reviewed (approved):
Looks good!

alecbcs commented on var/spack/repos/builtin/packages/flux-sched/package.py:
```diff
@@ -186,6 +186,8 @@ def setup_run_environment(self, env):
 
 class CMakeBuilder(CMakeBuilder):
     def cmake_args(self):
+        if not os.path.exists(os.path.join(self.stage.source_path, "flux-sched.ver")):
+            return [f"-DFLUX_SCHED_VER={self.spec.version}"]
```
```suggestion
        if not os.path.exists(os.path.join(self.stage.source_path, "flux-sched.ver")):
            return [self.define("FLUX_SCHED_VER", self.spec.version)]
```

alecbcs reviewed (dismissed):
Optional suggestion to clean up the syntax a bit and conform to the latest recommended syntax. Otherwise looks good to me. @trws let me know if you'd prefer to merge as is.

trws committed:
Update var/spack/repos/builtin/packages/flux-sched/package.py

Co-authored-by: Alec Scott <hi@alecbcs.com>

trws commented:
Committed the tweak, doing an update, then could you enable auto-merge @alecbcs?

alecbcs reviewed (dismissed):
Looks good to me

trws committed:
flux-sched: ensure we set the version for older releases

the version file isn't enough for some releases and needs it explicitly
set even if the file is there, this will address that

trws commented:
Sorry to do this again, we found a case where it wasn't applying to back versions and had to poke it.  Could you recheck @alecbcs?

alecbcs commented on var/spack/repos/builtin/packages/flux-sched/package.py:
```diff
@@ -186,6 +186,9 @@ def setup_run_environment(self, env):
 class CMakeBuilder(CMakeBuilder):
     def cmake_args(self):
+        ver_in_src = os.path.exists(os.path.join(self.stage.source_path, "flux-sched.ver"))
+        if self.spec.satisfies("@:0.33") or not ver_in_src:
```
```suggestion
        # flux-sched before v0.33 does not correctly set the version even when the file is present.
        if self.spec.satisfies("@:0.33") or not ver_in_src:
```

No worries, I'd probably suggest documenting the edge case here so that someone doesn't change it later on by accident.

trws committed:
Update var/spack/repos/builtin/packages/flux-sched/package.py

Co-authored-by: Alec Scott <hi@alecbcs.com>

trws commented:
Looks like this all passed, but the approve got cleared by merging the comment @alecbcs.
